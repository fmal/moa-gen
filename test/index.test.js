'use strict';

const micro = require('micro');
const listen = require('test-listen');
const got = require('got');
const qs = require('query-string');

const serviceHandler = require('../src');

describe('moa generator service', () => {
  let service;
  let url;

  beforeAll(async () => {
    service = micro(serviceHandler);
    url = await listen(service);
  });

  afterAll(() => {
    service.close();
  });

  it('responds 200 with correct headers', async () => {
    const response = await got(url);

    expect(response.statusCode).toBe(200);
    expect(response.headers['content-type']).toBe('image/svg+xml');
    expect(response.headers['cache-control']).toBe('public,s-maxage=3600');
  });

  it('responds 400 when url is invalid', async () => {
    expect.assertions(1);

    const invalidUrl = `${url}/path`;
    try {
      await got(invalidUrl);
    } catch (error) {
      expect(error.response.statusCode).toEqual(400);
    }
  });

  it('correctly handles query params', async () => {
    const query = qs.stringify({
      size: '500',
      name: 'Excellence',
      receiver: 'Test'
    });
    const response = await got(`${url}?${query}`);
    expect(response.body).toMatchSnapshot();
  });
});
