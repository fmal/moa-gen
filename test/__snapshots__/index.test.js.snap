// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`moa generator service correctly handles query params 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<svg width=\\"500\\" height=\\"667\\" viewBox=\\"0 0 960 720\\" xmlns=\\"http://www.w3.org/2000/svg\\">
  <pattern id=\\"texture\\" height=\\"4\\" patternUnits=\\"userSpaceOnUse\\" width=\\"4\\">
    <path d=\\"M1 3h1v1H1zm2-2h1v1H3z\\" fill=\\"#9c92ac\\" opacity=\\".5\\"/>
  </pattern>10
  <g fill=\\"none\\" stroke=\\"#9c92ac\\" stroke-width=\\"6\\" transform=\\"translate(10 10)\\">
    <circle cx=\\"10\\" cy=\\"10\\" r=\\"10\\"/>
    <circle cx=\\"930\\" cy=\\"10\\" r=\\"10\\"/>
    <circle cx=\\"10\\" cy=\\"690\\" r=\\"10\\"/>
    <circle cx=\\"930\\" cy=\\"690\\" r=\\"10\\"/>
  </g>
  <path d=\\"M50 0h840a50 50 0 0 0 50 50v600a50 50 0 0 0-50 50H50a50 50 0 0 0-50-50V50A50 50 0 0 0 50 0z\\" fill=\\"url(#texture)\\" stroke=\\"#9c92ac\\" stroke-width=\\"8\\" transform=\\"translate(10 10)\\"/>
  <text fill=\\"#3d3847\\" font-family=\\"monospace\\" font-size=\\"52\\" text-anchor=\\"middle\\" word-spacing=\\"-10\\" x=\\"50%\\" y=\\"178\\">
    Certificate <tspan font-family=\\"Georgia, serif\\" font-size=\\"40\\" font-style=\\"italic\\">of</tspan> Excellence
  </text>
  <text fill=\\"#3d3847\\" font-family=\\"monospace\\" text-anchor=\\"middle\\" x=\\"50%\\" y=\\"266\\">
    <tspan font-size=\\"32\\">AWARDED TO</tspan><tspan dy=\\"84\\" font-size=\\"48\\" font-weight=\\"bold\\" x=\\"50%\\">Test</tspan>
  </text>
  <path d=\\"M300 390h420\\" stroke=\\"#9c92ac\\" stroke-dasharray=\\"2 8\\" stroke-linecap=\\"round\\" stroke-width=\\"6\\"/>
  <path d=\\"M510 616.1L480.7 630l-15.4-28-32.1-5.6 4.5-31.5L415 542l22.7-23-4.5-31.4 32.1-5.6 15.4-28 29.3 13.9 29.3-13.9 15.4 28 32.1 5.6-4.5 31.5L605 542l-22.7 23 4.5 31.4-32.1 5.6-15.4 28z\\" fill=\\"none\\" opacity=\\".9\\" stroke=\\"#9c92ac\\" stroke-linejoin=\\"round\\" stroke-width=\\"8\\"/>
  <path d=\\"M492 576h-17.6c-2.4 0-4.4-2-4.4-4.5v-40c0-2.5 2-4.5 4.4-4.5H492zm58-17.8c0 9.8-8 17.8-17.7 17.8H497v-48.9c4.9 0 8.8-4 8.8-8.9v-17.8c0-2.4 2-4.4 4.5-4.4h4.4c4.8 0 8.8 4 8.8 8.9v17.8h17.7c4.8 0 8.8 4 8.8 8.9z\\" fill=\\"#9c92ac\\"/>
</svg>"
`;
